diff --git a/Makefile.common b/Makefile.common
index 3277ff0..a4b3686 100644
--- a/Makefile.common
+++ b/Makefile.common
@@ -1070,6 +1070,10 @@ ifneq ($(HAVE_LAKKA_CANARY), "")
    DEFINES += -DHAVE_LAKKA_CANARY=\"${HAVE_LAKKA_CANARY}\"
 endif
 
+ifeq ($(HAVE_LAKKA_DEVBUILD), 1)
+   DEFINES += -DHAVE_LAKKA_DEVBUILD
+endif
+
 ifeq ($(HAVE_MENU_COMMON), 1)
    OBJ += menu/menu_setting.o \
           menu/menu_driver.o \
diff --git a/file_path_special.h b/file_path_special.h
index a80c0b6..cd01bdc 100644
--- a/file_path_special.h
+++ b/file_path_special.h
@@ -78,8 +78,10 @@ RETRO_BEGIN_DECLS
 #define FILE_PATH_CORE_THUMBNAILPACKS_URL "http://thumbnailpacks.libretro.com"
 #ifdef HAVE_LAKKA_CANARY
 #define FILE_PATH_LAKKA_URL HAVE_LAKKA_CANARY
-#elif HAVE_LAKKA_NIGHTLY
+#elif defined (HAVE_LAKKA_NIGHTLY)
 #define FILE_PATH_LAKKA_URL "http://nightly.builds.lakka.tv/.updater"
+#elif defined (HAVE_LAKKA_DEVBUILD)
+#define FILE_PATH_LAKKA_URL "http://nightly.builds.lakka.tv/.devbuild"
 #else
 #define FILE_PATH_LAKKA_URL "http://le.builds.lakka.tv"
 #endif
